libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFNetwork.h

```diff
@@ -28,12 +28,28 @@
  */
 
 #import <Foundation/Foundation.h>
+#import "SalesforceSDKConstants.h"
 
 @interface SFNetwork : NSObject
 
 typedef void (^SFDataResponseBlock) (NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error);
 
-@property (nonatomic, readwrite, assign) BOOL useBackground;
+@property (nonatomic, readonly, assign) BOOL useBackground;
+@property (nonatomic, readonly, strong, nonnull) NSURLSession *activeSession;
+
+/**
+ * Initializes this class with an ephemeral session configuration.
+ *
+ * @return Instance of this class.
+ */
+- (nonnull instancetype)initWithEphemeralSession;
```

Comment:
Designated initializer for each type of `NSURLSessionConfiguration` so that apps can ask for what they want instead of the SDK creating them all.